### ISSUE TITLE
Core Dump when creating CNN to learn over images

### ISSUE THREAD

**Issue by manishpatelUK** [deeplearning4j/deeplearning4j#8696]
#### Issue Description

Creating a fairly shallow CNN. Using a (semi-but-sensible) randomly generated CNN config, to the effect of:

- 11 layers, alternating between convoluted and subsampling
- kernel sizes at 5,5 (input layer) 2,2 (otherwise)
- stride at 1,1
- padding at 3,3
- simple output layer with loss function set to negative liklihood, activation set to softmax.
- I have this (from example) not sure if it's actually required:
`listBuilder.setInputType(InputType.convolutional(height, width, channels));`

Execution eventually ends in a core dump:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007ffd0992ca40, pid=6760, tid=0x0000000000001a48
#
# JRE version: Java(TM) SE Runtime Environment (8.0_161-b12) (build 1.8.0_161-b12)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.161-b12 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# C  [ntdll.dll+0x1ca40]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  T H R E A D  ---------------

Current thread (0x0000000002c09000):  JavaThread "main" [_thread_in_native, id=6728, stack(0x0000000002b00000,0x0000000002c00000)]

siginfo: ExceptionCode=0xc0000005, reading address 0xffffffffffffffff

Registers:
RAX=0x00000000000004bb, RBX=0x0000000001220000, RCX=0x0000000000000a0e, RDX=0x00000000000004bb
RSP=0x0000000002bfaeb0, RBP=0x0000000000000000, RSI=0x0000000000000000, RDI=0x00000000bf0004bb
R8 =0xffc00000ffbffff0, R9 =0x000000000000ec90, R10=0x0000000000000001, R11=0x00007ffd09a29500
R12=0x0000000001220150, R13=0x0000000000000ec9, R14=0xffc00000ffc00000, R15=0x0000000001222590
RIP=0x00007ffd0992ca40, EFLAGS=0x0000000000010207

Top of Stack: (sp=0x0000000002bfaeb0)
0x0000000002bfaeb0:   0000000000000004 0000000000000312
0x0000000002bfaec0:   0000000000000000 0000000000000000
0x0000000002bfaed0:   0000000001220000 00007ffd0992922b
0x0000000002bfaee0:   0000000100000001 0000000000000000
0x0000000002bfaef0:   0000000000000000 0000000000000000
0x0000000002bfaf00:   0000000000000000 0000000000000ec9
0x0000000002bfaf10:   0000000000000001 0000000062450422
0x0000000002bfaf20:   0000000031410240 00007ffd094b9960
0x0000000002bfaf30:   0000000000000028 0000000000000028
0x0000000002bfaf40:   0000000000000000 3f8000003f800000
0x0000000002bfaf50:   000000003f800000 00000000000003ff
0x0000000002bfaf60:   0000000031415c68 0000000031412b44
0x0000000002bfaf70:   000000002ea97790 00007ffd09929e8e
0x0000000002bfaf80:   0000000031415c78 0000000062460ba8
0x0000000002bfaf90:   0000000002bfb260 000000006cd7f0c0
0x0000000002bfafa0:   000000002ea978d0 00007ffd09929e8e 

Instructions: (pc=0x00007ffd0992ca40)
0x00007ffd0992ca20:   e9 aa fc ff ff 41 39 77 0c 0f 85 b1 0e 00 00 49
0x00007ffd0992ca30:   8b 47 30 4e 8b 34 f0 4d 3b e6 74 56 4d 8d 46 f0
0x00007ffd0992ca40:   41 8b 50 08 89 94 24 b8 01 00 00 39 73 7c 74 2e
0x00007ffd0992ca50:   8b bb 88 00 00 00 33 fa 89 bc 24 b8 01 00 00 0f 


Register to memory mapping:

RAX=0x00000000000004bb is an unknown value
RBX=0x0000000001220000 is an unknown value
RCX=0x0000000000000a0e is an unknown value
RDX=0x00000000000004bb is an unknown value
RSP=0x0000000002bfaeb0 is pointing into the stack for thread: 0x0000000002c09000
RBP=0x0000000000000000 is an unknown value
RSI=0x0000000000000000 is an unknown value
RDI=0x00000000bf0004bb is an unknown value
R8 =0xffc00000ffbffff0 is an unknown value
R9 =0x000000000000ec90 is an unknown value
R10=0x0000000000000001 is an unknown value
R11=0x00007ffd09a29500 is an unknown value
R12=0x0000000001220150 is an unknown value
R13=0x0000000000000ec9 is an unknown value
R14=0xffc00000ffc00000 is an unknown value
R15=0x0000000001222590 is an unknown value


Stack: [0x0000000002b00000,0x0000000002c00000],  sp=0x0000000002bfaeb0,  free space=1003k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.nd4j.nativeblas.Nd4jCpu$NativeOps.execCustomOp(Lorg/bytedeco/javacpp/PointerPointer;JLorg/bytedeco/javacpp/Pointer;)I+0
j  org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(Lorg/nd4j/linalg/api/ops/CustomOp;Lorg/nd4j/linalg/api/ops/OpContext;)[Lorg/nd4j/linalg/api/ndarray/INDArray;+89
j  org.nd4j.linalg.factory.Nd4j.exec(Lorg/nd4j/linalg/api/ops/CustomOp;Lorg/nd4j/linalg/api/ops/OpContext;)[Lorg/nd4j/linalg/api/ndarray/INDArray;+5
j  org.deeplearning4j.nn.layers.mkldnn.MKLDNNConvHelper.preOutput(Lorg/nd4j/linalg/api/ndarray/INDArray;Lorg/nd4j/linalg/api/ndarray/INDArray;Lorg/nd4j/linalg/api/ndarray/INDArray;[I[I[ILorg/deeplearning4j/nn/conf/layers/ConvolutionLayer$AlgoMode;Lorg/deeplearning4j/nn/conf/layers/ConvolutionLayer$FwdAlgo;Lorg/deeplearning4j/nn/conf/ConvolutionMode;[ILorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)Lorg/nd4j/linalg/api/ndarray/INDArray;+450
j  org.deeplearning4j.nn.layers.convolution.ConvolutionLayer.preOutput(ZZLorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)Lorg/nd4j/linalg/primitives/Pair;+507
j  org.deeplearning4j.nn.layers.convolution.ConvolutionLayer.preOutput4d(ZZLorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)Lorg/nd4j/linalg/primitives/Pair;+4
j  org.deeplearning4j.nn.layers.convolution.ConvolutionLayer.backpropGradient(Lorg/nd4j/linalg/api/ndarray/INDArray;Lorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)Lorg/nd4j/linalg/primitives/Pair;+328
j  org.deeplearning4j.nn.multilayer.MultiLayerNetwork.calcBackpropGradients(Lorg/nd4j/linalg/api/ndarray/INDArray;ZZZ)Lorg/nd4j/linalg/primitives/Pair;+555
j  org.deeplearning4j.nn.multilayer.MultiLayerNetwork.computeGradientAndScore()V+384
j  org.deeplearning4j.nn.multilayer.MultiLayerNetwork.computeGradientAndScore(Lorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)V+1
j  org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(Lorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)Lorg/nd4j/linalg/primitives/Pair;+13
j  org.deeplearning4j.optimize.solvers.StochasticGradientDescent.optimize(Lorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)Z+85
j  org.deeplearning4j.optimize.Solver.optimize(Lorg/deeplearning4j/nn/workspace/LayerWorkspaceMgr;)V+9
j  org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fitHelper(Lorg/nd4j/linalg/dataset/api/iterator/DataSetIterator;)V+541
j  org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(Lorg/nd4j/linalg/dataset/api/iterator/DataSetIterator;)V+2
(my own code ommittted)
```

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 1.0.0-beta4
* Platform information (OS, etc) Windows 10.0 , 64 bit Build 17134 (10.0.17134.753)
Java HotSpot(TM) 64-Bit Server VM (25.161-b12) for windows-amd64 JRE (1.8.0_161-b12), built on Dec 19 2017 17:52:25 by "java_re" with MS VC++ 10.0 (VS2010)

#### Additional Information

Where applicable, please also provide:

* Full log or exception stack trace (ideally in a Gist: gist.github.com)
* pom.xml file or similar (also in a Gist)

#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!

**Comment by AlexDBlack** [comment]
Best guess is that this is an invalid configuration that is slipping through validation.

First: I noticed that you're on 1.0.0-beta4. Upgrade to 1.0.0-beta6 and try again (no point us debugging this if it's already fixed).

If it still happens on 1.0.0-beta6, can you share the Java code for us to try and reproduce/debug this?

**Comment by manishpatelUK** [comment]
Thanks, yes it's probably a misconfig - upgrading gives me an error instead of a core dump. This is in evaluation phase rather than `fit()`. Will post code when I find what combination of params is causing it.

```
java.lang.IllegalStateException: Cannot perform evaluation with NaNs present in predictions: 2 NaNs present in predictions INDArray

	at org.nd4j.base.Preconditions.throwStateEx(Preconditions.java:641)
	at org.nd4j.base.Preconditions.checkState(Preconditions.java:286)
	at org.nd4j.evaluation.classification.Evaluation.eval(Evaluation.java:405)
	at org.nd4j.evaluation.BaseEvaluation.eval(BaseEvaluation.java:271)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.doEvaluationHelper(MultiLayerNetwork.java:3436)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.doEvaluation(MultiLayerNetwork.java:3384)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.evaluate(MultiLayerNetwork.java:3578)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.evaluate(MultiLayerNetwork.java:3488)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.evaluate(MultiLayerNetwork.java:3319)

```

**Comment by AlexDBlack** [comment]
Thanks for testing.
OK, so I'll close this then. Maybe it was a bug on our end in 1.0.0-beta4 (specific to MKLDNN or something).

NaNs could be due to bad tuning - i.e., learning rate too high. Or lack of normalization for input data. Or bad combination of loss function + activation function for output layer, for example.

I'll close this as resolved, if you can't work out the NaNs feel free to ask at https://community.konduit.ai/
